### DISCUSSION TITLE
Remote BIN files: deciphering them is nice, being able to load them would be even better

### DISCUSSION THREAD

**Discussion by DVoorhees1957** [GerryDazoo/Slinger#415] [General]
Although I've now figured out how to use RM (RemoteMaster) to decipher/understand the contents of remote BIN files, that doesn't seem like much of a victory in and of itself. It would be truly wonderful if we could figure out how to load a BIN file into a Slingbox. Unfortunately, the only explanation I've seen for doing this relies entirely on using the Slingplayer Desktop software (http://www.hifi-remote.com/wiki/index.php/How_to_Install_a_Custom_Remote_onto_the_Slingbox). Did anyone perchance do a packet capture of the BIN loading process prior to N9?

**Comment by GerryDazoo** [comment]
Yes, we did. Are you volunteering to reverse engineer the captures and write the code?

**Reply by DVoorhees1957** [reply]
I'm certainly happy to take a look :-) ... just don't expect too much! I'd say my Python coding abilities are only a 6 or so on a scale of 10 (I often rely on adapting snippets of code that already provide a similar function)... and I know almost nothing about the communication process between Slingplayer Desktop and a Slingbox.

**Reply by DVoorhees1957** [reply]
I'm in the process of downloading 500_setup_full.pcapng - I assume that's what I want to look at? And do you have any hint about the various phases of setup and how (where) they correspond to in terms of that file?

**Reply by GerryDazoo** [reply]
I was being a little sarcastic, I've been slogging through that file for a few weeks. I've got some notes somewhere. I'm getting old, every time I get back to it I've already forgotten what I knew.

**Reply by DVoorhees1957** [reply]
It's cool, I really am happy to help (but probably less-capable than you are). I just installed Wireshark, am about to reboot and then open the 500_setup_full file...

**Reply by DVoorhees1957** [reply]
@GerryDazoo Do you happen to remember which remote BIN file you were setting up the Slingbox 500 with? I thought maybe I could search for the matching hex contents in the packet capture (assuming it gets transferred from Slingplayer Desktop to the Slingbox without any translation)...

EDIT: OK, that's silly, I don't have any idea how to use Wireshark to isolate specific communication sequences between your PC (192.168.117.2) and the Slingbox 500 (192.168.117.110). I figured out how I can filter to find packets flowing between the two (i.e., "ip.addr==192.168.117.2 and ip.addr==192.168.117.110 and tcp.port==5201"), but I don't know what I'm looking for.

**Reply by abarberenaCPDS** [reply]
Hey @DVoorhees1957 👋 , if you are trying to do a web capture, you may use [Fiddler](https://www.telerik.com/download/fiddler). If keep trying with Wireshark, refer to [Wireshark Masterclass](https://www.youtube.com/watch?v=-HDpYR_QSFw&list=PLW8bTPfXNGdC5Co0VnBK1yVzAwSSphzpJ&index=5).

**Reply by GerryDazoo** [reply]
I'll take a look. BTW here's a dump of the interesting IPs in the file
site2ip = {
 'apollo.slingbox.com' :     '24.222.184.185',
 'autoupdate.sling.com' : '24.222.184.195',
 'sparcs.sling.com' :        '67.148.153.220',
 'myip.sling.com' :          '24.222.184.169',
 'analytics.sling.com' :     '67.148.153.116',
 'mdconfig.sling.com' :      '13.225.195.15',
 'relay.sling.com' :         '8.39.227.143',  
 'msg.sling.com' :           '67.148.153.136',
 'ngsb-setup.slingbox.com' : '24.222.184.178',
 'mdregistry.sling.com' :    '67.148.153.215',
 'secure.sling.com' :        '67.148.153.111',
 'imasdk.googleapis.com' :    '172.217.13.106',
 'setupservices.slingbox.com': '67.148.153.158',
 'SB500' :                    '192.168.117.110'
'Desktop':                 '192.168.117.2'
}

Some other preamble about the file. It's big but most of it is video streams the DT code setup while it was configuring the box. You can filter those out and save the file. It makes wireshark much move responsive when applying other display filters.

I was capturing everything coming out of my PC so there is some noise especially at the start.
If you are writing code it's doesn't doesn't have to be python I think anything that can run on windows would be OK. There is a SlingboxSDK.dll that could be leveraged. And one called SlingPlayerAX that was used to make the slinger_utility that rb231 wrote. There was also another little player example on github which I now cannot find that used the sdk and "blingsox" on github that uses tyoescript code. We also have all or most of the Java code that the DT uses to do useful stuff. 
PS. I don't have the remote bin code off the top of my head but it was a Motorola PVR dcx3400M I think. Have a look at packet 443.

**Reply by DVoorhees1957** [reply]
Is this the "Blingsox" github project that you were referencing: https://github.com/Nomkid/Blingsox ? There's no documentation, so it's kinda hard to tell :-)

**Reply by GerryDazoo** [reply]
Yup. That's it.

**Comment by DVoorhees1957** [comment]
@GerryDazoo I found 2 references to possibly matching BIN files in the packet data: S1116_PL.bin (slingbox.fandom says this is an SBC Dish satellite remote) and C1283_PL.bin (slingbox.fandom says this is a Comcast cable remote). Do either of those ring a bell? The 1st is at frame 463, and the 2nd (and the more likely one) is at frames 186844 and 186857...

**Reply by GerryDazoo** [reply]
On another note: In the slingplayerax dll that rbf121 used, There is this little honey

	def UploadIRBinFile(self, fileName=defaultNamedNotOptArg):
		'IR bin file will be uploaded into box, current input must not be tuner.'
		return self._oleobj_.InvokeTypes(9, LCID, 1, (24, 0), ((8, 1),),fileName
			)

**Reply by GerryDazoo** [reply]
To isolate a particular tcp/http stream. Select a packet in the stream and click Analyze/Follow ....

**Reply by DVoorhees1957** [reply]
The UploadIRBinFile routine seems promising (at least based on the name)!

**Comment by abarberenaCPDS** [comment]
Long ago, I was helped by Alan Richey, he pointed me to a similar post (hi-fi). I am adding the links to this thread for reference:

- How to create custom remote upgrade: [https://slingbox.fandom.com/wiki/How_to_create_custom_remote_upgrade](https://slingbox.fandom.com/wiki/How_to_create_custom_remote_upgrade)
- Slingbox BIN files: [http://www.hifi-remote.com/forums/dload.php?action=category&cat_id=87](http://www.hifi-remote.com/forums/dload.php?action=category&cat_id=87)
- Remote images: [http://autoupdate.sling.com/remotedb/Preview/](http://autoupdate.sling.com/remotedb/Preview/)
